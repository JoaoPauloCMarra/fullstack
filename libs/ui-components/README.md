# UI Components Library for Web and Mobile