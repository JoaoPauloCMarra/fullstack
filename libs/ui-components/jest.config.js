module.exports = {
  displayName: 'ui-components',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { cwd: __dirname, configFile: './babel-jest.config.json' }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ui-components',
  setupFilesAfterEnv: ['@testing-library/jest-native/extend-expect'],
};
