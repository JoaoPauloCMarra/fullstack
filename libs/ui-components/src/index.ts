export { default as Center } from './components/Center';
export { default as Title } from './components/Title';

export default {};
