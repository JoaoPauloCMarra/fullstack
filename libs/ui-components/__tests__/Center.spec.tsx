import React from 'react';
import 'react-native-web';
import renderer from 'react-test-renderer';

import { Center } from '@nx-fullstack/ui-components';

test('renders correctly', () => {
  const tree = renderer.create(<Center />);
  expect(tree.toJSON()).toMatchSnapshot();
});
