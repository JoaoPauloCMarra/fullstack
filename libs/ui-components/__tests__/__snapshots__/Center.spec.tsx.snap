// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxPack": "center",
      "WebkitFlex": 1,
      "WebkitJustifyContent": "center",
      "alignItems": "center",
      "backgroundColor": "rgba(255,255,255,1.00)",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "msFlex": "1 1 0%",
      "msFlexAlign": "center",
      "msFlexPack": "center",
      "width": "100%",
    }
  }
/>
`;
